tests/run/cpython_capi.pyx

```diff
@@ -5,18 +5,36 @@ from cpython cimport mem
 from cpython.pystate cimport PyGILState_Ensure, PyGILState_Release, PyGILState_STATE
 
 
+cdef short _assert_calloc(short * s, int n):
+    """Assert array ``s`` of length ``n`` is zero and return 3."""
+    s[0] += 1
+    s[n - 1] += 3
+    assert not s[0] and not s[n - 1]
+    for i in range(1, n - 1):
+        assert not s[i]
+    return s[n - 1]
+
+
 def test_pymalloc():
     """
     >>> test_pymalloc()
     3
     """
+    cdef short i
+    cdef short* s = <short*> mem.PyMem_Calloc(10, sizeof(*s))
```

Comment:
Actually, the bug is here. you probably meant to say `s[0]`. But I'd advise against that, because you're only just declaring it right in the same statement. Just use `sizeof(short)` (also below).

Reply:
Should be resolved by 52677e2—un-resolve this thread if it isn't.